docker/katana/Dockerfile

```diff
@@ -0,0 +1,58 @@
+FROM --platform=$BUILDPLATFORM rust:1.70 as katana-builder
+
+RUN apt-get update && apt-get install -y git
+
+WORKDIR /app
+
+RUN git clone https://github.com/dojoengine/dojo.git
+
+WORKDIR /app/dojo
+
+# add targets
+RUN rustup target add x86_64-unknown-linux-gnu
+RUN rustup target add aarch64-unknown-linux-gnu
+
+# install compilers
+RUN apt-get -y install gcc-x86-64-linux-gnu
+RUN apt-get -y install gcc-aarch64-linux-gnu
+
+# build for both targets
+RUN cargo build --package katana --release --target=x86_64-unknown-linux-gnu --config target.x86_64-unknown-linux-gnu.linker=\"x86_64-linux-gnu-gcc\"
+RUN cargo build --package katana --release --target=aarch64-unknown-linux-gnu --config target.aarch64-unknown-linux-gnu.linker=\"aarch64-linux-gnu-gcc\"
+
+FROM python:3.9.13-slim-bullseye 
+
+HEALTHCHECK --interval=30s --timeout=30s --start-period=5s --retries=3 \
+    CMD if [ "$(cat /root/.kakarot_deploy_status)" -eq 0 ]; then exit 0; else exit 1; fi
+
+
+COPY --from=katana-builder /app/dojo/target/x86_64-unknown-linux-gnu/release/katana  /app/katana/x86_64-unknown-linux-gnu/release/katana 
+COPY --from=katana-builder /app/dojo/target/aarch64-unknown-linux-gnu/release/katana /app/katana/aarch64-unknown-linux-gnu/release/katana 
+COPY ./docker/katana/scripts /app/scripts
+
+WORKDIR /app
+
+RUN apt-get update && apt-get install -y build-essential coreutils curl libgmp3-dev
+
+RUN chmod +x scripts/run.sh
+RUN chmod +x scripts/wait_and_deploy.sh
+
+COPY ./docker/katana/cairo/requirements.txt ./requirements.txt
+
+RUN pip3 install -r requirements.txt
+
+# install poetry
+RUN curl -sSL https://install.python-poetry.org | python3 -
+
+ENV PATH="$PATH:/root/.local/bin"
+
+COPY . ./kakarot
```

Comment:
it's usually better to first copy the requirements only (here the pyproject.toml and the poetry.lock) to leverage caching in `docker build`.

Then, copy the whole project, otherwise each modified file will invalidate the (usually long) install deps step.

Reply:
Makes sense! Thanks for this info.

I have made the change [here.]([feat: group all COPY togther in the Dockerfile](https://github.com/kkrt-labs/kakarot/pull/594/commits/9bc4e11c247626cac56198c657247b7b0994934d))

Reply:
maybe it wasn't clear. You need to 

```
# first only the deps files
COPY poetry.lock poetry.lock
COPY pyproject.toml pyproject.toml

poetry install

# then the whole project
COPY . .
```

Reply:
Got it! 

Thanks, I have implemented [here](https://github.com/kkrt-labs/kakarot/pull/594/commits/c8c960eba543cf861124467f8ee5fbdb124c0ab5).